### DISCUSSION TITLE
Best way to get current gas price.

### DISCUSSION THREAD

**Discussion by duralsh** [ava-labs/coreth#690] [Q&A]
we are working on a feature where our frontend displays an adjustable gas price pop up. The idea is super simple, we try to detect if the fees are higher than common 25 gwei and if it is we give our users the option to set their gas fees manually, for testing purposes we set this to 40 gwei.

The issue is, rpc nodes never seem to return anything other than 25 gwei. I have written a simple script to fetch gas fee data with ethers.js and over the course of 8 hours, it has only gotten 25 gwei and nothing else. Is this some hardcoded value on the rpc node level ? 

Do you have any suggestions on how to fetch the current gas price ? We need a reliable way of tracking the last block's gas price so we can adjust accordingly.

**Comment by darioush** [comment]
Could you please clarify which RPC endpoint you are using to obtain your data and which network you are querying?

**Reply by fffbbbccc** [reply]
we are using the public avalanche c-chain rpc url to get the c-chain fee data.

**Comment by fffbbbccc** [comment]
A related question to that, is 25 gwei as base fee a static value ? If not, what would be the trigger for it to increase ? For example, does block space usage (gas used / 15 M total gas limit per block) has any impact on the base fee ? 

My understanding is, 25 gwei is the base minimum to have a tx included in a block eventually. Paying anything above that would put your tx in front of the queue but if the block space utilization is low (like 10 percent etc), it actually doesnt make sense to pay anything more than 25 gwei. Does this make sense ? 

PS: 
Lets imagine this scenario : 
MaxFeePerGas : 75 gwei
MaxPriorityFee : 20 gwei
BaseFee : 25 gwei 

Here a user would pay 45 gwei but there is still a margin of 30 gwei, remaining from MaxFeePerGas - (MaxPriorityFee + BaseFee ). This 30 gwei is refunded but if 25 gwei base fee was static, no one would have needed this margin at all, doing BaseFee + MaxPriorityFee = MaxFeePerGas would have sufficed. So i suppose BaseFee is not static and might increase. My question again is, how is this calculated/estimated ?

**Reply by darioush** [reply]
BaseFee on the C-Chain is calculated using an algorithm ([see here](https://github.com/ava-labs/coreth/blob/e73e655356b7df05193e50d5aa56866b3be68e04/consensus/dummy/dynamic_fees.go#L42)) inspired by EIP1559
There is a target of gas usage for the past N seconds, and the fee increases / decreases to match this target based on demand.
The algorithm operates within min/max boundaries, and the current min baseFee for C-chain mainnet is 25 navax.

To facilitate better use of blockspace, this is set to reduce to 1 navax in the Etna upgrade, which is already active on testnet.
https://github.com/avalanche-foundation/ACPs/tree/main/ACPs/125-basefee-reduction
